modules/dev-server/main.tf

```diff
@@ -401,7 +401,11 @@ exec consul agent -server \
   -hcl='cert_file = "/consul/${var.datacenter}-server-consul-0.pem"' \
   -hcl='key_file = "/consul/${var.datacenter}-server-consul-0-key.pem"' \
   -hcl='auto_encrypt = {allow_tls = true}' \
+%{if local.is_consul_1_14_plus~}
+  -hcl='ports { https = 8501, grpc_tls = 8503 }' \
+%{else~}
   -hcl='ports { https = 8501, grpc = 8502 }' \
+%{endif~}
```

Comment:
A note on this: I didn't need any changes to the Consul client agent config, because we do not enable TLS locally within the task. The client agent is configured with [`ports.grpc = 8502`](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/8986f2c418cd389638fc9ea44f0a1857f61b5988/modules/mesh-task/templates/consul_agent_defaults.hcl.tpl#L16) and [mesh-init](https://github.com/hashicorp/consul-ecs/blob/76f957ed59f7fa0fcf48f000628b0cbc2957c256/subcommand/mesh-init/command.go#L127) configures Envoy with that port (8502).